src/main/java/seedu/academyassist/model/AcademyAssist.java

```diff
@@ -173,11 +173,11 @@ public boolean personDuplicateClass(Set<Subject> subjects, Person student) {
      * Adds {@code subject} to {@code person}
      */
     public void addSubjectsToPerson(Set<Subject> subjects, Person person) {
-        persons.remove(person);
+        Person updatedPerson = person;
         for (Subject s: subjects) {
-            person = person.addSubject(s);
+            updatedPerson = updatedPerson.addSubject(s);
         }
-        persons.add(person);
+        persons.setPerson(person, updatedPerson);
```

Comment:
Good that you use setPerson for updated Person to avoid deleting and adding again!